log/fatal.go

```diff
@@ -0,0 +1,64 @@
+package log
+
+import (
+	"fmt"
+)
+
+// Common errors that can happen on node startup.
+var (
+	ErrMalformedConfig    = newFatalError("ERR_MALFORMED_CONFIG", "config file is malformed: %v")
```

Comment:
let's use a code only for errors that are meaningful for smapp. for example ERR_START_PROFILING doesn't make sense

Reply:
@dshulyak 
If we're talking only about solving the current issue — that's right.
However, I think that the homogeneous shape of fatal errors is a good idea. And if we took this is for the standard for fatal errors, then developers do not have to think anymore "Do somebody need an error code for custom handling of such errors?".

Reply:
no, fatal error is not something that should be added in the first place. any error can lead to fatal exit. code is needed only if smapp has a special handling for some of the codes.

Reply:
@dshulyak What are the disadvantages of having errors like `ERR_START_PROFILING`? With the homogeneous implementation, they are cheap to add, so smapp can decide whether or not it needs them

Reply:
disadvantage is that now every error must be have a code. which is a nonsense

Reply:
Why? It must not. If an error doesn't have a code, it works as before this PR

Reply:
this is pretty insignificant problem that will impose requirements for writing errors across whole application. we can cover couple of errors that smapp wants to present in a nice way (like clock issue), but in the general case it will have to present information that it is available in the log.

honestly i think that even clock thing doesn't require custom presentation. just show what is in the log

Reply:
> Why? It must not. If an error doesn't have a code, it works as before this PR

then what was the reason to add those weird wrappers in the first place? let them work as before

Reply:
to clarify. you did that because you went over the source and added code to errors that could lead to fatal exit. however this list is not exhaustive. so some errors that will lead to exit will have code and some not. tomorrow i will add another error in some package and i will need a code. this is clearly nonsense.

Reply:
> this is pretty insignificant problem that will impose requirements for writing errors across whole application

I don't understand why you think that those requirements will be imposed. 

> honestly i think that even clock thing doesn't require custom presentation

Do you mean the wrappers or anything else by "custom presentation"?

> then what was the reason to add those weird wrappers in the first place? let them work as before

What do you disagree with? The way the wrappers are implemented or adding more context to errors other that clock error?

> tomorrow i will add another error in some package that could lead to an exit and i will need a code

You will be able to create a usual error like we used to do, the error code just won't be shown in logs and the smapp wouldn't know how to classify the error. What are you suggesting instead? Not to add exit codes? But the result would be the same as in case you described

Reply:
i suggest adding a code for a specific subset of errors that will be known to smapp. currently it will be only clock error. otherwise smapp should just present an error that is specified in fatal log.

as for implementation please see example above, it allows to encode descriptive object without args